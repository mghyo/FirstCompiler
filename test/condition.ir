#start_function main
void main():
int-list: $t0, i, j
float-list: 
  assign, i, 5, 
  brleq, i, 42, label0
  assign, j, 0
  goto, label1, ,
label0:
  assign, j, 1, 
  goto, label1, , 
label1:
  call, printi, j
  return,,,
#end_function main
