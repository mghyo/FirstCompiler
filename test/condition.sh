#!/bin/bash

set -e

./phase2 test/condition.ir $1

diff out.s test/condition.$1.s

spim -f out.s > tmp

diff tmp test/condition.out

