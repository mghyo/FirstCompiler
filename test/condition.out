Loaded: /usr/share/spim/exceptions.s
1